ext/sockets/sendrecvmsg.c

```diff
@@ -302,9 +302,15 @@ PHP_FUNCTION(socket_cmsg_space)
 		return;
 	}
 
-	if (entry->var_el_size > 0 && n > (zend_long)((ZEND_LONG_MAX - entry->size -
-			CMSG_SPACE(0) - 15L) / entry->var_el_size)) {
-		/* the -15 is to account for any padding CMSG_SPACE may add after the data */
+	size_t rem_size = ZEND_LONG_MAX - entry->size;
+	size_t n_max = entry->var_el_size > 0 ? rem_size / entry->var_el_size : 0;
+	size_t size = entry->size + n * entry->var_el_size;
+	size_t total_size = CMSG_SPACE(size);
+
+	if (entry->var_el_size > 0
```

Comment:
It would be better to make `if (entry->var_el_size > 0) {}` an outer if and move the code above inside it, to avoid the repeated and rather confusing check inside n_max.

Reply:
Done, and forced push, this indeed makes the code less confusing.